packages/core/src/tools/serialisation/storedContextManager.ts

```diff
@@ -0,0 +1,89 @@
+import { computeBytesCount } from '../utils/byteUtils'
+import { ONE_SECOND } from '../utils/timeUtils'
+import { throttle } from '../utils/functionUtils'
+import type { ContextManager } from './contextManager'
+import { createContextManager } from './contextManager'
+import type { Context } from './context'
+import type { CustomerDataType } from './serializationConstants'
+
+const CONTEXT_STORE_KEY_PREFIX = '_dd_c'
+const STORAGE_POLL_DELAY = ONE_SECOND
+export function createStoredContextManager(
+  customerDataType: CustomerDataType,
+  computeBytesCountImpl = computeBytesCount
+): ContextManager {
+  const contextManager = createContextManager(customerDataType, computeBytesCountImpl)
+
+  const synchronizeWithStorage = () => contextManager.setContext(read(customerDataType))
+  const dumpToStorage = () => store(customerDataType, contextManager.getContext())
+  const { throttled: synchronizeWithStorageThrottled } = throttle(synchronizeWithStorage, STORAGE_POLL_DELAY, {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏You could use the `storage` event instead of polling and reading synchronously the value from the storage.

I think it would be much easier: read the context from the store at init, then update the context every time we receive a `storage` event.